lib/suppressions.cpp

```diff
@@ -173,6 +173,15 @@ std::string Suppressions::addSuppressionLine(const std::string &line)
     Suppressions::Suppression suppression;
     if (std::getline(lineStream, suppression.errorId, ':')) {
         if (std::getline(lineStream, suppression.fileName)) {
+            const std::string::size_type hashpos = suppression.fileName.find("#");
+            const std::string::size_type cppcommentpos = suppression.fileName.find("//");
+            std::string::size_type endpos = std::min(hashpos, cppcommentpos);
+            if (endpos != std::string::npos) {
+                while (endpos > 0 && std::isspace(suppression.fileName[endpos-1])) {
+                    endpos--;
+                }
+                suppression.fileName = suppression.fileName.substr(0, endpos);
```

Comment:
nit; I prefer `suppression.fileName.erase(endpos);`

Reply:
You are not very nitpicky.  I fixed this, and even added some unit tests, but then realizing that the suppression comment handling would only work for lines containing filenames.

I think this would be confusing to users, so I've rewritten this to strip the comment before creating the istringstream at all, but this takes me back to using substr.

Please check the latest version and provide me with more nits.